src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -170,6 +170,82 @@ def workerCleanup(workerCleanupInfo):
         # TODO: logic here for cleanup flag
         shutil.rmtree(workflowDir)
 
+    @staticmethod
+    def probeUsedDisk(workflowID, workDir=None):
+        '''
+        Used to probe the used disk space (on the disk mounting workDir) minus the space used by
+        Toil. Subtracting this from the total disk size gives the batchsystem an estimate of how
+        much space is available for Toil. On Mesos, this can be reserved for a non-Toil role.
+
+        :param str workflowID: Unique ID for this workflow
+        :param str workDir: value for --workDir passed by the user.
+        :return: Amount of disk used by non-Toil purposes
+        '''
+        workflowDir = getToilWorkflowDir(workflowID, workDir)
+        toilDiskUsed = AbstractBatchSystem.getWorkflowDirSize(workflowDir)
+        diskStats = os.statvfs(workflowDir)
+        # Total Disk used = block size * (total blocks - free blocks)
+        diskUsed = diskStats.f_frsize * (diskStats.f_blocks - diskStats.f_bavail)
+        # Disk used for non-toil purposes = Total disk used - Disk used by toil (workDir)
+        nonToilDiskUsed = diskUsed - toilDiskUsed
+        return nonToilDiskUsed
+
+    @staticmethod
+    def getWorkflowDirSize(workflowDir):
+        '''
+        This job will walk through workflowDir and return the cumulative filesize in bytes of all
+        the files in the directory and its subdirectories.
+
+        :param workflowDir: path to the workflow working directory
+        :return: cumulative size in bytes of all files in the directory
+        '''
+        total_size = 0
+        # The value from running stat on each linked file is equal. To prevent the same file
+        # from being counted multiple times, we save the inodes of files that have more than one
+        # nlink associated with them.
+        linkedFileInodes = set()
+        for dirpath, dirnames, filenames in os.walk(workflowDir):
+            folderSize = AbstractBatchSystem._getDirSize(dirpath, filenames, linkedFileInodes)
+            total_size += folderSize
+        return total_size
+
+    @staticmethod
+    def _getDirSize(dirpath, filenames, linkedFileInodes):
+        '''
+        Aux for getWorkDirSize. returns the size of the directory including all linked files
+        provided their inode was not in linkedFileInodes
+
+        :param str dirpath: Dir path
+        :param list filenames: Files in the directory
+        :param set linkedFileInodes: set of inodes of file objects that have multiple links and that
+                                     have already been acccounted for.
+        :return: Size of the directory
+        '''
+        folderSize = 0
+        for f in filenames:
+            fp = os.path.join(dirpath, f)
+            fileStats = os.stat(fp)
+            if fileStats.st_nlink > 1:
+                if fileStats.st_ino not in linkedFileInodes:
+                    folderSize += fileStats.st_blocks * 512
```

Comment:
Can you extract 512 into a global variable called unixBlockSize and in a comment above explain why it's ok to hard-code it?